lib/protoflow-blocks/src/blocks/text/split_string.rs

```diff
@@ -85,19 +84,8 @@ impl Block for SplitString {
     fn execute(&mut self, runtime: &dyn BlockRuntime) -> BlockResult {
         runtime.wait_for(&self.input)?;
 
-        let regex = if self.delimiter.is_empty() {
-            None
-        } else {
-            Regex::new(&self.delimiter).ok()
-        };
-
         while let Some(input) = self.input.recv()? {
-            let outputs: Vec<String> = if let Some(ref regex) = regex {
-                regex.split(&input).map(|s| s.to_string()).collect()
-            } else {
-                input.split(&self.delimiter).map(|s| s.to_string()).collect()
-            };
-
+            let outputs: Vec<String> = input.split(&self.delimiter).map(|s| s.to_string()).collect();
```

Comment:
No need to collect these into memory, could just iterate and output them directly without allocating more memory. But that's an optimization we can address later.

Reply:
fixed. create a new PR